internal/routing/acquire.go

```diff
@@ -15,11 +16,27 @@ func acquire(w http.ResponseWriter, r *http.Request, ls *lockservice.SimpleLockS
 		return
 	}
 
-	desc := &lockservice.SimpleDescriptor{
-		FileID: string(body),
+	var req interface{}
+	err = json.Unmarshal(body, &req)
+
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
 	}
 
-	err = ls.Acquire(desc)
+	data := req.(map[string]interface{})
```

Comment:
Why's there a map here? Are you assuming there can be descriptors of multiple types?
This probably has to engineered differently. Assume there's a single descriptor instance now and then we can have a batching feature later on.

Reply:
https://golang.org/pkg/encoding/json/#Unmarshal

map[string]interface{}, for JSON objects

I feel passing `lockservice.Descriptors` is damn clean on the client side but I guess it's a little messy on the server's side, in terms of reconstructing what was sent. 

I haven't even used the `Request` struct that I had created since there is no way to access the ID of the descriptor because `d` is of type `lockservice.Descriptors`. Any ideas on what clean way of resolving this issue is?

Reply:
No longer relevant since this logic is no longer being used.